managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1483,17 +1483,16 @@ public synchronized void createComplete(int rc, final LedgerHandle lh, Object ct
         } else {
             log.info("[{}] Created new ledger {}", name, lh.getId());
             ledgers.put(lh.getId(), LedgerInfo.newBuilder().setLedgerId(lh.getId()).setTimestamp(0).build());
```

Comment:
Maybe we can also move this line to metadata persistent success? and then we don't need to remove it when store metadata failed.

Reply:
Fixed. @mattisonchao

Reply:
I got @merlimat comment, sorry.